src/discordtipbot/statistics.cr

```diff
@@ -0,0 +1,43 @@
+struct Statistics
+  DB.mapping({
+    transactions: {
+      type: Int64,
+      key:  "transaction_count",
+    },
+    total: {
+      type: BigDecimal,
+      key:  "transaction_sum",
+    },
+    tips: {
+      type: BigDecimal,
+      key:  "tip_sum",
+    },
+    soaks: {
+      type: BigDecimal,
+      key:  "soak_sum",
+    },
+    rains: {
+      type: BigDecimal,
+      key:  "rain_sum",
+    },
+  })
+
+  @@ttl : Time::Span = 1.minutes
+
+  @@last = Time.utc_now
+
+  def self.get(db : DB::Database)
+    update(db)
+    Statistics.from_rs(db.query("SELECT * FROM statistics")).last
+  end
+
+  def self.update(db : DB::Database)
+    return unless Time.utc_now > @@last + @@ttl
+    db.exec("REFRESH MATERIALIZED VIEW statistics")
+    @@last = Time.utc_now
+  end
+
+  def last
```

Comment:
This is a `class_getter`

Reply:
If I use a `class_getter` I get following error:
```
undefined method 'last' for Statistics
```

Reply:
@z64 not sure if I'm just misunderstanding something here.

Reply:
```
class_getter foo

# expands to:
def self.foo
  @@foo
end
```